example/App.js

```diff
@@ -32,11 +32,64 @@ export default class App extends Component {
     this._onStateChanged = TrackPlayer.addEventListener('playback-state', (data) => {
       PlayerStore.playbackState = data.state;
     })
+    
+    
+    this._updateIcyHeader = TrackPlayer.addEventListener('icy-header-update', (data) => {
+      console.log("Update IceCast Header!", data)
+    })
+    this._updateIcyMetaData = TrackPlayer.addEventListener('icy-metadata-update', this.icyMetaDataShow(data))
   }
 
   componentWillUnmount() {
     this._onTrackChanged.remove()
     this._onStateChanged.remove()
+
+    this._updateIcyHeader.remove()
+    this._updateIcyMetaData.remove()
+  }
+
+  findGetParameter(url) {
+    var result = {};
+    url.substr(1).split("&").forEach(function (item) {
+      const tmp = item.split("=");
+      result[decodeURIComponent(tmp[0])] = decodeURIComponent(tmp[1]);
+    });
+    return result;
+  }
+
+  icyMetaDataShow = async ({streamTitle, streamUrl}) =>{   
+    console.log("Update IceCast MetaData!", streamTitle); 
+    const curTrack = await TrackPlayer.getCurrentTrack();
+
+    if(curTrack != null) {
+      const Track = await TrackPlayer.getTrack(curTrack);
+      
+      let music = 'Unknow';
+      let singer = '...';
+      streamTitle = streamTitle.trim();
+      if(streamTitle != '' && streamTitle.toUpperCase().indexOf('ADVERT:') != 0) {
+          const corte = streamTitle.indexOf("-");
+          if(corte > 0) {
+              music = streamTitle.substr(0, corte).trim();
+              singer = streamTitle.substr(corte+1, streamTitle.length-(corte+1)).trim();
+          } else {
+              music = streamTitle.substr(0, streamTitle.length).trim();
+              singer = '';
+          }
+      } else {
+        // Advert metadata
+      }
+
+      Track.title = music;
+      Track.artist = singer;
+      TrackPlayer.updateMetadata(Track);
+      
+      TrackStore.title = Track.title;
+      TrackStore.artist = Track.artist;
+
+      const params = this.findGetParameter(streamUrl);
+      console.log("MetadataUpdate, Url params:", params, "Stream Title:", streamTitle);
+    }
```

Comment:
I don't think this type of code fits with the example app. It's meant to be a simple app showing the basic usage of the module.